runtime/lua/vim/lsp/util.lua

```diff
@@ -1795,8 +1795,18 @@ function M.locations_to_items(locations, offset_encoding)
       local row = pos.line
       local end_row = end_pos.line
       local line = lines[row] or ''
-      local col = M._str_byteindex_enc(line, pos.character, offset_encoding)
-      local end_col = M._str_byteindex_enc(lines[end_row] or '', end_pos.character, offset_encoding)
+      local end_line = lines[end_row] or ''
+      ---@type boolean, integer, integer
+      local ok, col, end_col
+      ok, col = pcall(_str_byteindex_enc, line, pos.character, offset_encoding)
+      if not ok then
+        col = vim.fn.strcharlen(line) - 1
+      end
+      ok, end_col = pcall(_str_byteindex_enc, end_line, end_pos.character, offset_encoding)
+      if not ok then
+        end_col = vim.fn.strcharlen(end_line) - 1
+      end
```

Comment:
The type annotation can be avoided by assigning the `local` declaration.

```suggestion
      local ok, col = pcall(_str_byteindex_enc, line, pos.character, offset_encoding)
      if not ok then
        col = vim.fn.strcharlen(line) - 1
      end
      local ok2, end_col = pcall(_str_byteindex_enc, end_line, end_pos.character, offset_encoding)
      if not ok2 then
        end_col = vim.fn.strcharlen(end_line) - 1
      end
```